### PR TITLE
agura - add google search test

### PR DESCRIPTION
**Pull request by Gura-Andriy**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class GoogleHomePage {
+    protected static WebDriver driver = new ChromeDriver();;
+
+    public GoogleHomePage openGoogleSearch() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java]: rename open*Page()

File: src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        driver.get("https://www.google.com");
+        driver.manage().window().maximize();
+        return this;
+    }
+
+    public void doSearch(String inputText) {
+        WebElement searchInputField = driver.findElement(By.xpath("//input[@name='q']"));
+        searchInputField.sendKeys(inputText);
+        searchInputField.sendKeys(Keys.ENTER);
+    }
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java:44]: this should return new page object

File: src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public boolean checkLinkHasText(int indexOfLink, String link) {
+        if (link.contains(
+                Stream.of(driver
+                        .findElements(By.xpath("//h3[@class='LC20lb DKV0Md']")))
+                        .map(element -> element
+                                .get(indexOfLink)
+                                .getText())
+                        .collect(Collectors
+                                .joining()))) return true;
+        else return false;
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java]: this method is way too complicated, you should return search results and check them in test class, not in page object

File: src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java [hunk 4]
```diff
@@ -1,20 +0,0 @@
-package com.softserveinc.ita.nromanchuk;
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java:1]: DO NOT CHANGE someone else's code

File: src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        System.setProperty("webdriver.chrome.driver", ".C:\\chromedriver.exe");
+        googlePage.openGoogleSearch();
+        googlePage.doSearch(searchText);
+        System.out.println(resultPage.checkLinkHasText(indexOfLink, checkingText));
+    }
+
+    @Test
+    public void testing() {
+        testingSearch(0, "funny kitten", "funnykitten");
+        testingSearch(0, "smartphone", "wikipedia");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java]: separate into two different tests

File: src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java [hunk 6]
```diff
@@ -1,14 +0,0 @@
-package com.softserveinc.ita.pageobjects_task.nromanchuk;
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java:1]: DO NOT CHANGE someone else's code

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 7]
```diff
@@ -31,14 +31,20 @@
...
-            <version>${chrome.driver.version}</version>
+            <artifactId>selenium-support</artifactId>
+            <version>3.141.59</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.3.0</version>
+            <scope>compile</scope>
```
Comment (toleksyn) [inline pom.xml]: duplication

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java (hunk 4 above)
Comment (Gura-Andriy) [reply src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java:1]: I accidentally did it and did not notice. I will be more attentive

File: src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java (hunk 6 above)
Comment (Gura-Andriy) [reply src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java:1]: I accidentally did it and did not notice. I will be more attentive

Comment (anna-gura) [reply src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java:1]: I accidentally did it and did not catch sight of. I will be more attentive

File: src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java (hunk 5 above)
Comment (Gura-Andriy) [reply src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java]: I accidentally did it and did not catch sight of. I will be more attentive

File: src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java (hunk 4 above)
Comment (anna-gura) [reply src/main/java/com/softserveinc/ita/nromanchuk/GoogleHomePage.java:1]: I accidentally did it and did not catch sight of. I will be more attentive

File: pom.xml [hunk 8]
```diff
@@ -31,14 +31,14 @@
         </dependency>
         <dependency>
             <groupId>org.seleniumhq.selenium</groupId>
-            <artifactId>selenium-chrome-driver</artifactId>
-            <scope>test</scope>
-            <version>${chrome.driver.version}</version>
+            <artifactId>selenium-support</artifactId>
```
Comment (toleksyn) [inline pom.xml:34]: not needed

File: src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package com.softserveinc.ita.gura;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class GoogleHomePage {
+    protected static WebDriver driver = new ChromeDriver();;
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleHomePage.java]: move to test runner class

File: src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package com.softserveinc.ita.gura;
+
+import org.openqa.selenium.WebDriver;
+import static com.softserveinc.ita.gura.GoogleHomePage.driver;
+
+public class GoogleResultPage {
+    public static WebDriver getDriver() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java:8]: this belongs to testrunner

File: src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void testingSearchWithSearchTextFunnyKitten() {
+        testingContainOfLink(0, "funny kitten", "funnykitten");
+    }
+
+    @Test
+    public void testingSearchWithSearchTextSmartphone() {
+        testingContainOfLink(0, "smartphone", "wikipedia");
+    }
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/pageobjects_task/gura/GoogleTest.java:79]: do not over complicate, stick to the task
test should end with an assert

File: src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java (hunk 6 above)
Comment (toleksyn) [reply src/test/java/com/softserveinc/ita/pageobjects_task/nromanchuk/GoogleTest.java:1]: revert

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class GoogleResultPage {
+    public static WebDriver getDriver() {
+        return TestRunner.driver;
+    }
+
+    public GoogleImagePage goToImagePage() {
+        WebElement imagesBottom = getDriver().findElement(By.xpath("//*[@id=\"hdtb-msb\"]/div[1]/div/div[2]/a"));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java]: do not use \"
refactor

File: src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+        return new GoogleImagePage(imagesBottom.getAttribute("href"));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/gura/GoogleResultPage.java:14]: no arguments to page objects

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 14]
```diff
@@ -31,14 +31,14 @@
         </dependency>
```
Comment (toleksyn) [inline pom.xml:31]: into dev branch, not master

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**